### PR TITLE
feat: support parsing `html` via `@html-eslint/parser` natively

### LINKED ISSUE #651: feat: support other HTML related ESLint plugins
> We can also support other HTML related ESLint plugins like https://github.com/yeonjuan/html-eslint

_Originally posted by @JounQin in https://github.com/prettier/eslint-plugin-prettier/issues/621#issuecomment-1866692220_

Would this be possible?

### PR DESCRIPTION
**Pull request by Logicer16**
See https://github.com/prettier/eslint-plugin-prettier/pull/621#issuecomment-1866692220

Closes #651 

~~Currently blocked by https://github.com/yeonjuan/html-eslint/issues/184 for the parser's meta object.~~

Please let me know if there's anything else I should include.

### CODE REVIEW

File: worker.js [hunk 1]
```diff
@@ -111,6 +109,18 @@ runAsWorker(
           }
           break;
         }
+        case 'html': {
+          // it could be processed by `eslint-plugin-html` or correctly parsed by `@html-eslint/parser`
+          if (
+            (typeof parserMeta !== 'undefined' &&
+              parserMeta.name !== '@html-eslint/parser') ||
+            (typeof parserPath === 'string' &&
+              !/([\\/])@html-eslint\/parser\1/.test(parserPath))
```
Comment (JounQin) [inline worker.js:118]:
```suggestion
              !/([\\/])@html-eslint\1parser\1/.test(parserPath))
```

**Review by JounQin — CHANGES_REQUESTED** [review]

File: worker.js (hunk 1 above)
Comment (Logicer16) [reply worker.js]: Good find!

**Review by JounQin — APPROVED** [review]

File: .changeset/rich-swans-itch.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"eslint-plugin-prettier": patch
```
Comment (JounQin) [inline .changeset/rich-swans-itch.md:2]:
```suggestion
"eslint-plugin-prettier": minor
```

**Comment by JounQin** [comment]
test

**Pull request merged**